### ISSUE TITLE
`MD041` flags H1 HTML tags containing line breaks

### ISSUE THREAD

**Issue by 5ouma** [rvben/rumdl#152]
```
README.md:1:1: [MD041] First line in file should be a level 1 heading
```

In a markdown document:

```md
<h1>
Some text
</h1>
```

**Comment by rvben** [comment]
Fix released! :) Thanks for raising the issue, @5ouma !

**Comment by 5ouma** [comment]
Hello, @rvben,

Thank you for addressing this issue!
I'm sorry to address the edge case after you've fixed it.

```
README.md:1:1: [MD041] First line in file should be a level 1 heading
```

In a Markdown document:

```md
<h1>
  <div>
    <img
      href="https://example.com/image.png"
      alt="Example Image"
    />
    <a
      href="https://example.com"
    >Example Project</a>
  </div>
</h1>
```

The core issue seems like this:

- h1
  - any-tag
    - any-tag with more than two properties separated by new lines
    - any-tag with more than one property separated by new lines

**Comment by rvben** [comment]
Thanks for the feedback, @5ouma. I've improved the html parsing in a couple of rules, including for this case.

**Comment by 5ouma** [comment]
@rvben 

Thank you again very much for fixing the issue.
I'm sorry to tell you again, but I found that this will make an error even on v0.0.179.

```md
<h1>
  <div>
    <img
      href="https://example.com/image.png"
      alt="Example Image"
    />
    <a
      href="https://example.com"
    >Example Project</a>
    <span>Documentation</span>
  </div>
</h1>
```

This will occur when I want to use a picture tag.

```md
<h1>
  <picture>
    <source
      srcset="https://example.com/light.png"
      media="(prefers-color-scheme: light)"
    />
    <source
      srcset="https://example.com/dark.png"
      media="(prefers-color-scheme: dark)"
    />
    <img src="https://example.com/default.png" />
  </picture>
</h1>
```
